sway-core/src/decl_engine/wrapper.rs

```diff
@@ -189,6 +232,21 @@ impl From<ty::TyEnumDeclaration> for (DeclWrapper, Span) {
 }
 
 impl DeclWrapper {
+    fn discriminant_value(&self) -> u8 {
+        match self {
+            DeclWrapper::Unknown => 0,
+            DeclWrapper::Function(_) => 1,
+            DeclWrapper::Trait(_) => 2,
+            DeclWrapper::TraitFn(_) => 3,
+            DeclWrapper::ImplTrait(_) => 4,
+            DeclWrapper::Struct(_) => 5,
+            DeclWrapper::Storage(_) => 6,
+            DeclWrapper::Abi(_) => 7,
+            DeclWrapper::Constant(_) => 8,
+            DeclWrapper::Enum(_) => 9,
+        }
+    }
```

Comment:
Can [std::mem::discriminant](https://doc.rust-lang.org/std/mem/fn.discriminant.html) be used here, since `Discriminant` implements `Hash`?

Reply:
Another common pattern is to add another `enum DeclKind { Unknown, Function, Trait, etc... }` that can be used for distinguishing between decl kinds, implement `Hash` etc.

That said I've never seen discriminant used in the wild - looks like an interesting option!